pkg/vsphere/vsan/host_vsan_internal_system.go

```diff
@@ -0,0 +1,159 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vsan
+
+import (
+	"context"
+	"encoding/json"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/methods"
+	//	"github.com/vmware/govmomi/vim25/mo"
+	//	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+	//	"github.com/vmware/vic/pkg/trace"
+)
+
+const (
+	DomObject  = "DOM_OBJECT"
+	LsomObject = "LSOM_OBJECT"
+	Policy     = "POLICY"
+	Disk       = "DISK"
+)
+
+type VSANExtAttrs struct {
+	Path string `json:"Object path"`
+}
+
+// TODO: HostVsanInternalSystem should be moved to govmomi
+type HostVsanInternalSystem struct {
+	object.Common
+	cl *vim25.Client
+}
+
+func NewHostVsanInternalSystem(c *vim25.Client, ref types.ManagedObjectReference) *HostVsanInternalSystem {
+	m := HostVsanInternalSystem{
+		Common: object.NewCommon(c, ref),
+		cl:     c,
+	}
+
+	return &m
+}
+
+// QueryVsanObjectUuidsByFilter returns vsan DOM object uuids by filter
+func (m HostVsanInternalSystem) QueryVsanObjectUuidsByFilter(ctx context.Context, uuids []string, limit int32, version int32) ([]string, error) {
+	req := types.QueryVsanObjectUuidsByFilter{
+		This:    m.Reference(),
+		Uuids:   uuids,
+		Limit:   limit,
+		Version: version,
+	}
+
+	res, err := methods.QueryVsanObjectUuidsByFilter(ctx, m.cl, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	if res == nil {
+		return nil, nil
+	}
+	return res.Returnval, nil
+}
+
+// QueryVsanObjects returns vsan detail object information
+func (m HostVsanInternalSystem) QueryVsanObjects(ctx context.Context, uuids []string) (string, error) {
+	req := types.QueryVsanObjects{
+		This:  m.Reference(),
+		Uuids: uuids,
+	}
+
+	res, err := methods.QueryVsanObjects(ctx, m.cl, &req)
+	if err != nil {
+		return "", err
+	}
+
+	if res == nil {
+		return "", nil
+	}
+	return res.Returnval, nil
+}
+
+// GetVsanObjExtAttrs returns vsan extended vsan attributes, including object path. This API can be slow based on API doc
+func (m HostVsanInternalSystem) GetVsanObjExtAttrs(ctx context.Context, uuids []string) (map[string]VSANExtAttrs, error) {
+	var ret map[string]VSANExtAttrs
+
+	req := types.GetVsanObjExtAttrs{
+		This:  m.Reference(),
+		Uuids: uuids,
+	}
+
+	res, err := methods.GetVsanObjExtAttrs(ctx, m.cl, &req)
+	if err != nil {
+		return ret, err
+	}
+
+	if res == nil {
+		return ret, nil
+	}
+	log.Debugf("Returned attributes: %s", res.Returnval)
```

Comment:
hmm.. actually it's just one debug information during my development.
This should be in govmomi, and I didn't see any such kind of debug information in govmomi code.